src/Paramol/SqlClient/TSqlDecimalValue.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+
+namespace Paramol.SqlClient
+{
+    /// <summary>
+    ///     Represents a T-SQL DECIMAL parameter value.
+    /// </summary>
+    public class TSqlDecimalValue : IDbParameterValue
+    {
+        /// <summary>
+        ///     The single instance of this value.
+        /// </summary>
+        public static readonly TSqlDecimalValue Instance = new TSqlDecimalValue();
+
+        private readonly decimal _value;
+        private readonly byte _precision;
+        private readonly byte _scale;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public TSqlDecimalValue()
+        {
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="precision"></param>
+        /// <param name="scale"></param>
+        public TSqlDecimalValue(decimal value, byte precision = 18, byte scale = 0)
```

Comment:
I know I'm nitpicking at this stage, but I prefer specific types such as `TSqlDecimalPrecision` and `TSqlDecimalScale` instead of primitives such as `byte` (moves the range checking into those types - with an implicit operator to go back to byte). Other than that everything is looking good. I'll have a closer look tonight, introduce those missing types and ship.